File: Graphs/DeleteVertex.java
// File Name: DeleteVertex.java

import java.util.*;

public class DeleteVertex {

    static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    static int vertices;

    static void deleteVertex(int v) {
        if (v >= vertices) {
            System.out.println("Invalid vertex!");
            return;
        }

        // Remove all edges to this vertex
        for (int i = 0; i < vertices; i++) {
            graph.get(i).remove(Integer.valueOf(v));
        }

        // Remove the vertex list
        graph.remove(v);

        // Adjust vertex numbers in adjacency lists
        for (int i = 0; i < graph.size(); i++) {
            for (int j = 0; j < graph.get(i).size(); j++) {
                if (graph.get(i).get(j) > v) {
                    graph.get(i).set(j, graph.get(i).get(j) - 1);
                }
            }
        }

        vertices--;
        System.out.println("Vertex " + v + " deleted successfully.");
    }

    static void displayGraph() {
        System.out.println("\nAdjacency List:");
        for (int i = 0; i < graph.size(); i++) {
            System.out.println(i + " -> " + graph.get(i));
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of vertices: ");
        vertices = sc.nextInt();

        for (int i = 0; i < vertices; i++) {
            graph.add(new ArrayList<>());
        }

        while (true) {
            System.out.println("\n1. Delete Vertex\n2. Display Graph\n3. Exit");
            System.out.print("Enter your choice: ");
            int choice = sc.nextInt();

            switch (choice) {
                case 1:
                    System.out.print("Enter vertex to delete: ");
                    int v = sc.nextInt();
                    deleteVertex(v);
                    break;
                case 2:
                    displayGraph();
                    break;
                case 3:
                    System.exit(0);
                default:
                    System.out.println("Invalid choice");
            }
        }
    }
}

File: Graphs/delete_vertex.py
# File Name: delete_vertex.py

graph = []
vertices = int(input("Enter number of vertices: "))

for i in range(vertices):
    graph.append([])

def delete_vertex(v):
    global vertices
    if v >= vertices:
        print("Invalid vertex!")
        return

    # Remove all edges pointing to v
    for adj in graph:
        if v in adj:
            adj.remove(v)

    # Remove the vertex itself
    graph.pop(v)

    # Adjust remaining vertex numbers
    for adj in graph:
        for i in range(len(adj)):
            if adj[i] > v:
                adj[i] -= 1

    vertices -= 1
    print(f"Vertex {v} deleted successfully.")

def display_graph():
    print("\nAdjacency List:")
    for i in range(len(graph)):
        print(f"{i} -> {graph[i]}")

while True:
    print("\n1. Delete Vertex")
    print("2. Display Graph")
    print("3. Exit")
    choice = int(input("Enter your choice: "))

    if choice == 1:
        v = int(input("Enter vertex to delete: "))
        delete_vertex(v)
    elif choice == 2:
        display_graph()
    elif choice == 3:
        break
    else:
        print("Invalid choice")
